src/extensions/unit/unitext_hooks.cpp

```diff
@@ -680,6 +681,285 @@ DECLARE_PATCH(_UnitClass_Per_Cell_Process_AutoHarvest_Assign_Harvest_Mission_Pat
 }
 
 
+/**
+ *  Helper function.
+ *  Creates a unit based on an already existing unit.
+ *  Returns the new unit if successful, otherwise null.
+ *
+ *  @author: Rampastring
+ */
+UnitClass* Create_Transform_Unit(UnitClass* this_ptr) {
+
+    UnitTypeClassExtension* unittypeext = Extension::Fetch<UnitTypeClassExtension>(this_ptr->Class);
+
+    UnitClass* newunit = reinterpret_cast<UnitClass*>(unittypeext->TransformsInto->Create_One_Of(this_ptr->House));
+    if (newunit == nullptr) {
+
+        /**
+         *  Creating the new unit failed! Re-mark our occupation bits and return false.
+         */
+        return nullptr;
+    }
+
+    // Try_To_Deploy copies the tag this way at 0x0065112C
+    if (this_ptr->Tag != nullptr) {
+        newunit->Attach_Tag(this_ptr->Tag);
+        this_ptr->Tag->AttachCount--;
+        this_ptr->Tag = nullptr;
+    }
+
+    newunit->ActLike = this_ptr->ActLike;
```

Comment:
This chunk can be moved to its own member function in UnitClassExtension, perhaps `void Copy_State_From(UnitClass *unit)`?.